modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/api/IgniteSqlImpl.java

```diff
@@ -68,4 +96,25 @@ public Statement createStatement(String query) {
     public StatementBuilder statementBuilder() {
         return new StatementBuilderImpl();
     }
+
+    @Override
+    public void start() {
+        executor.scheduleWithFixedDelay(
+                () -> {
+                    for (SessionImpl session : sessions.values()) {
+                        if (session.expired()) {
+                            session.closeAsync();
+                        }
+                    }
+                },
+                SESSION_EXPIRE_CHECK_PERIOD,
+                SESSION_EXPIRE_CHECK_PERIOD,
+                TimeUnit.MILLISECONDS
+        );
+    }
+
+    @Override
+    public void stop() throws Exception {
+        executor.shutdownNow();
```

Comment:
Should we close sessions on stop?
```suggestion
        executor.shutdownNow();
        sessions.values().foreach(SessionImpl::closeAsync);
        sessions.clear();
```

Reply:
done